assets/src/state.ts

```diff
@@ -0,0 +1,75 @@
+import { Dispatch as ReactDispatch } from "react"
+import { Route, RouteId } from "./skate.d"
+
+export interface State {
+  routes: null | Route[]
+  selectedRouteIds: RouteId[]
+}
+
+export const initialState: State = {
+  routes: null,
+  selectedRouteIds: [],
+}
+
+interface SetRoutesAction {
+  type: "SET_ROUTES"
+  payload: {
+    routes: Route[]
+  }
+}
+
+export const setRoutes = (routes: Route[]): SetRoutesAction => ({
+  type: "SET_ROUTES",
+  payload: { routes },
+})
+
+interface SelectRouteAction {
+  type: "SELECT_ROUTE"
+  payload: {
+    routeId: RouteId
+  }
+}
+
+export const selectRoute = (routeId: RouteId): SelectRouteAction => ({
+  type: "SELECT_ROUTE",
+  payload: { routeId },
+})
+
+interface DeselectRouteAction {
+  type: "DESELECT_ROUTE"
+  payload: {
+    routeId: RouteId
+  }
+}
+
+export const deselectRoute = (routeId: RouteId): DeselectRouteAction => ({
+  type: "DESELECT_ROUTE",
+  payload: { routeId },
+})
+
+type Action = SetRoutesAction | SelectRouteAction | DeselectRouteAction
+
+export type Dispatch = ReactDispatch<Action>
+
+export const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case "SET_ROUTES":
+      return {
+        ...state,
+        routes: action.payload.routes,
+      }
+    case "SELECT_ROUTE":
+      return {
+        ...state,
+        selectedRouteIds: [...state.selectedRouteIds, action.payload.routeId],
```

Comment:
Is it worth it to ensure that duplicate values aren't entered here (or are "unique"-ified away)? Or is that overkill since our interface effectively prevents it?

Reply:
I considered that. The failure mode is that you'd see multiple ladders for the same route, which I think is fine. And if there were a way in the interface to re-add a route, that'd be a bigger problem, and I'd rather not try to quietly sweep it under the rug here.

Reply:
Fair enough